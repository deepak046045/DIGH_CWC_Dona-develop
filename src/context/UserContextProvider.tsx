'use client';

import React, {
  Dispatch,
  ReactNode,
  SetStateAction,
  createContext,
  useMemo,
  useState,
} from 'react';
import {
  NavigationOption,
  ungatedNavigationOptions,
} from '@/models/NavigationOptions';
import { Appointment } from '@/models/Appointment';
import { AlertType } from '@/models/AlertType';
import mockAppointments from '@/mocks/mockAppointments';
import { CarePartner } from '@/models/CarePartner';
import { mockCarePartners } from '@/mocks/mockCarePartners';

type UserContextDataType = {
  isLoggedIn: boolean;
  setIsLoggedIn: Dispatch<SetStateAction<boolean>>;
  activeView: NavigationOption;
  setActiveView: Dispatch<NavigationOption>;
  alert: AlertType;
  setAlert: Dispatch<SetStateAction<AlertType>>;
  appointments: Appointment[];
  setAppointments: Dispatch<SetStateAction<Appointment[]>>;
  carePartners: CarePartner[];
  setCarePartners: Dispatch<SetStateAction<CarePartner[]>>;
};

export const UserContext = createContext<UserContextDataType>({
  isLoggedIn: false,
  setIsLoggedIn: () => {},
  activeView: ungatedNavigationOptions[0],
  setActiveView: () => {},
  alert: { isOpen: false, message: '', level: '' },
  setAlert: () => {},
  appointments: mockAppointments,
  setAppointments: () => {},
  carePartners: mockCarePartners,
  setCarePartners: () => {},
});

type Props = {
  children: ReactNode;
};

export function UserContextProvider({ children }: Props) {
  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
  const [activeView, setActiveView] = useState<NavigationOption>(
    ungatedNavigationOptions[0]
  );
  const [alert, setAlert] = useState<AlertType>({
    isOpen: false,
    message: '',
    level: '',
  });
  const [appointments, setAppointments] =
    useState<Appointment[]>(mockAppointments);
  const [carePartners, setCarePartners] =
    useState<CarePartner[]>(mockCarePartners);

  const values = useMemo(
    () => ({
      isLoggedIn,
      setIsLoggedIn,
      activeView,
      setActiveView,
      alert,
      setAlert,
      appointments,
      setAppointments,
      carePartners,
      setCarePartners,
    }),
    [isLoggedIn, activeView, alert, appointments, carePartners]
  );

  return <UserContext.Provider value={values}>{children}</UserContext.Provider>;
}
