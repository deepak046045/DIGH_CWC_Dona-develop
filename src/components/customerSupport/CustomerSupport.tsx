'use client';

import React, { useContext } from 'react';
import Image from 'next/image';
import { LdsButton, LdsHorizontalRule } from '@elilillyco/ux-lds-react';
import layout from '@/scss/themes/layout.module.scss';
import buttons from '@/scss/themes/buttons.module.scss';
import { UserContext } from '@/context/UserContextProvider';
import styles from './CustomerSupport.module.scss';

export default function CustomerSupport() {
  const { isLoggedIn } = useContext(UserContext);

  const customerSupportSection: JSX.Element[] = [];

  if (!isLoggedIn) {
    customerSupportSection.push(
      <div className={`${styles.background}`}>
        <div
          className={`${styles.customerSupportSignUpSection} ${layout.gutter}`}
        >
          <div className={styles.customerSupportSignUpSectionHeader}>
            Let us assist you on your journey. Sign up on this website to
          </div>
          <div className={styles.customerSupportSignUpSectionIconGroup}>
            <div className={styles.customerSupportSignUpSectionIcon}>
              <Image
                src="/icons/circle-calendar-icon.svg"
                alt="CalendarIcon"
                width="60"
                height="60"
              />
              <div className={styles.customerSupportSignUpSectionText}>
                Personalize a calendar of your appointments
              </div>
            </div>
            <div className={styles.customerSupportSignUpSectionIcon}>
              <Image
                src="/icons/care-partners-icon.svg"
                alt="CarePartnersIcon"
                width="60"
                height="60"
              />
              <div className={styles.customerSupportSignUpSectionText}>
                Assign care partners to assist you along your journey
              </div>
            </div>
            <div className={styles.customerSupportSignUpSectionIcon}>
              <Image
                src="/icons/appointment-reminders-icon.svg"
                alt="AppointmentRemindersIcon"
                width="60"
                height="60"
              />
              <div className={styles.customerSupportSignUpSectionText}>
                Set up appointment reminders for you and your care partners
              </div>
            </div>
          </div>
          <LdsButton
            classes={`${buttons.primary}`}
            data-testid="sign-up-button"
          >
            Sign up now
          </LdsButton>
        </div>
      </div>
    );
  }

  customerSupportSection.push(
    <div className={`${styles.customerSupportSection} ${layout.gutter}`}>
      <div className={styles.customerSupportSectionLeft}>
        <div className={styles.customerSupportSectionLeftHeaderText}>
          Talk to a human
        </div>
        <div className={styles.customerSupportHorizontalRule}>
          <LdsHorizontalRule />
        </div>
        <div className={styles.customerSupportSectionLeftSubheaderText}>
          If you have other questions about your medication, please contact us!
        </div>
      </div>
      <div className={styles.customerSupportSectionRight}>
        <div className={styles.customerSupportSectionPhone}>
          <Image
            src="/icons/phone-icon.svg"
            alt="PhoneIcon"
            width="35"
            height="37"
          />
          <div className={styles.customerSupportSectionPhoneText}>
            1-844-825-8966
          </div>
        </div>
        <div className={styles.customerSupportSectionHours}>
          <Image
            src="/icons/purple-hours-icon.svg"
            alt="HoursIcon"
            width="35"
            height="35"
          />
          <div className={styles.customerSupportSectionHoursText}>
            Mon - Fri | 8:00 AM - 10:00 PM Eastern
          </div>
        </div>
      </div>
    </div>
  );

  return customerSupportSection;
}
