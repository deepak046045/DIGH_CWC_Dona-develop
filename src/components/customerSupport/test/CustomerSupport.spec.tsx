import React from 'react';
import { UserContext } from '@/context/UserContextProvider';
import { render, screen } from '@testing-library/react';
import CustomerSupport from '../CustomerSupport';

jest.mock('crypto', () => ({ randomUUID: () => '123456789' }));

describe('CustomerSupport', () => {
  it('should render page when logged in', () => {
    const mockValue = {
      isLoggedIn: true,
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <CustomerSupport />
      </UserContext.Provider>
    );

    expect(screen.queryByTestId('sign-up-button')).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render the page when not logged in', () => {
    const mockValue = {
      isLoggedIn: false,
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <CustomerSupport />
      </UserContext.Provider>
    );

    expect(screen.queryByTestId('sign-up-button')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
