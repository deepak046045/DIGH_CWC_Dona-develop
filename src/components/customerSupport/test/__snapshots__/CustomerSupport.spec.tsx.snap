// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerSupport should render page when logged in 1`] = `
<div>
  <div
    class="customerSupportSection gutter"
  >
    <div
      class="customerSupportSectionLeft"
    >
      <div
        class="customerSupportSectionLeftHeaderText"
      >
        Talk to a human
      </div>
      <div
        class="customerSupportHorizontalRule"
      >
        <lds-hr />
      </div>
      <div
        class="customerSupportSectionLeftSubheaderText"
      >
        If you have other questions about your medication, please contact us!
      </div>
    </div>
    <div
      class="customerSupportSectionRight"
    >
      <div
        class="customerSupportSectionPhone"
      >
        <img
          alt="PhoneIcon"
          data-nimg="1"
          decoding="async"
          height="37"
          loading="lazy"
          src="/icons/phone-icon.svg"
          style="color: transparent;"
          width="35"
        />
        <div
          class="customerSupportSectionPhoneText"
        >
          1-844-825-8966
        </div>
      </div>
      <div
        class="customerSupportSectionHours"
      >
        <img
          alt="HoursIcon"
          data-nimg="1"
          decoding="async"
          height="35"
          loading="lazy"
          src="/icons/purple-hours-icon.svg"
          style="color: transparent;"
          width="35"
        />
        <div
          class="customerSupportSectionHoursText"
        >
          Mon - Fri | 8:00 AM - 10:00 PM Eastern
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CustomerSupport should render the page when not logged in 1`] = `
<div>
  <div
    class="background"
  >
    <div
      class="customerSupportSignUpSection gutter"
    >
      <div
        class="customerSupportSignUpSectionHeader"
      >
        Let us assist you on your journey. Sign up on this website to
      </div>
      <div
        class="customerSupportSignUpSectionIconGroup"
      >
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="CalendarIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/circle-calendar-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Personalize a calendar of your appointments
          </div>
        </div>
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="CarePartnersIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/care-partners-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Assign care partners to assist you along your journey
          </div>
        </div>
        <div
          class="customerSupportSignUpSectionIcon"
        >
          <img
            alt="AppointmentRemindersIcon"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/icons/appointment-reminders-icon.svg"
            style="color: transparent;"
            width="60"
          />
          <div
            class="customerSupportSignUpSectionText"
          >
            Set up appointment reminders for you and your care partners
          </div>
        </div>
      </div>
      <button
        class="lds-button primary"
        data-testid="sign-up-button"
        type="button"
      >
        Sign up now
      </button>
    </div>
  </div>
  <div
    class="customerSupportSection gutter"
  >
    <div
      class="customerSupportSectionLeft"
    >
      <div
        class="customerSupportSectionLeftHeaderText"
      >
        Talk to a human
      </div>
      <div
        class="customerSupportHorizontalRule"
      >
        <lds-hr />
      </div>
      <div
        class="customerSupportSectionLeftSubheaderText"
      >
        If you have other questions about your medication, please contact us!
      </div>
    </div>
    <div
      class="customerSupportSectionRight"
    >
      <div
        class="customerSupportSectionPhone"
      >
        <img
          alt="PhoneIcon"
          data-nimg="1"
          decoding="async"
          height="37"
          loading="lazy"
          src="/icons/phone-icon.svg"
          style="color: transparent;"
          width="35"
        />
        <div
          class="customerSupportSectionPhoneText"
        >
          1-844-825-8966
        </div>
      </div>
      <div
        class="customerSupportSectionHours"
      >
        <img
          alt="HoursIcon"
          data-nimg="1"
          decoding="async"
          height="35"
          loading="lazy"
          src="/icons/purple-hours-icon.svg"
          style="color: transparent;"
          width="35"
        />
        <div
          class="customerSupportSectionHoursText"
        >
          Mon - Fri | 8:00 AM - 10:00 PM Eastern
        </div>
      </div>
    </div>
  </div>
</div>
`;
